data/meterpreter/meterpreter.py

```diff
@@ -479,7 +479,7 @@ def get_packet_tcp(self):
 			pkt_length -= 8
 			packet = bytes()
 			while len(packet) < pkt_length:
-				packet += self.socket.recv(4096)
+				packet += self.socket.recv(pkt_length - len(packet))
```

Comment:
Will this handle the case when the packet arrives in multiple segments and you have to read more than once? I think it might need to track a little more state, e.g.:

https://github.com/rapid7/metasploit-framework/pull/4411/files#diff-bffa20e884ab4d5ba2cb72987ae65257L479